eventHandlers.js

```diff
@@ -0,0 +1,527 @@
+      handleUpDownArrowKeys(event);
+    }
+  }
+}
+
+function handleKeyupEvent(event) {
+  const target = event.target;
+  
+  if (target.matches('.SSN')) {
+    parseSSN(event);
+  }
+  
+  if (target.matches('input[type="tel"]')) {
+    parsePhoneNumber(event);
+  }
+
+  if (target.matches('label input:not([type="radio"]):not([type="checkbox"]), label textarea')) {
+    handleInputEvent(event);
+  }
+}
+
+function handleBlurFocusoutEvent(event) {
+  const target = event.target;
+  
+  if (target.matches('input[type="text"], input[type="number"], input[type="email"], input[type="tel"], input[type="date"], input[type="month"], input[type="time"], textarea, select')) {
+    textboxinput(target);
+    target.setAttribute("style", "size: 20 !important");
+  }
+}
+
+function handleInputEvent(event) {
+  const target = event.target;
+  
+  if (target.matches('input[type="text"], textarea')) {
+    const label = target.closest('label');
+    if (label) {
+      const radioCB = document.getElementById(label.htmlFor) || label.querySelector('input[type="radio"], input[type="checkbox"]');
+      if (radioCB && (radioCB.type === 'radio' || radioCB.type === 'checkbox')) {
+        // Check or uncheck the radio or checkbox based on the text input length.
+        target.value.length > 0 ? radioCB.checked = true : radioCB.checked = false;
+        radioAndCheckboxUpdate(radioCB);
+        target.dataset.lastValue = target.value;
+        textboxinput(target);  // Ensure the text input is saved
+      }
+    }
+
+    // // Handle "Other" text inputs
+    // const responseContainer = target.closest('.response');
+    // if (responseContainer) {
+    //   debouncedHandleOtherTextInputKeyPress(event);
+    // }
+  }
+}
+
+function handleSubmitEvent(event) {
+  const target = event.target;
+
+  if (target.matches('.question') || target.closest('.question')) {
+      stopSubmit(event);
+  }
+}
+
+// Close the modal and focus on the question text.
+// Re-build the question text and focus management for screen readers.
+function closeModalAndFocusQuestion(event) {
+  const modal = document.getElementById('softModal');
+  const isWindowClick = event.target === modal;
+  const isButtonClick = ['close', 'modalCloseButton', 'modalContinueButton'].includes(event.target.id);
+
+  if (isWindowClick || isButtonClick) {
+    modal.style.display = 'none';
+
+    // Find the active question
+    const activeQuestion = document.querySelector('.question.active');
+
+    if (activeQuestion) {
+      const isModalClose = true;
+      manageAccessibleQuestionInit(activeQuestion.querySelector('fieldset') || activeQuestion, isModalClose);
+    }
+  }
+}
+
+// // Function to handle "Other" text input key press
+// function handleOtherTextInputKeyPress(event) {
+//   const responseTarget = event.target.closest('.response');
+//   const checkboxOrRadioEle = responseTarget?.querySelector('input[type="checkbox"], input[type="radio"]');
+
+//   if (checkboxOrRadioEle) {
```

Comment:
needed?

Reply:
I don't think this is needed based on changes I made to the base radio/checkbox handlers, but I'll need to retest in VoiceOver and JAWS (they behave differently) to confirm after everything is re-scoped...I'd like to keep it in for now and remove it in the cleanup pass.